libbeat/docs/processors-using.asciidoc

```diff
@@ -1246,3 +1247,167 @@ set to `true`, this condition will be ignored.
 `restricted_fields`:: (Optional) By default, the `process.env` field is not
 output, to avoid leaking sensitive data. If `restricted_fields` is `true`, the
 field will be present in the output.
+
+[[processor-script]]
+=== Script Processor
+
+experimental[]
+
+The script processor executes Javascript code to process an event. The processor
+uses a pure Go implementation of ECMAScript 5.1 and has no external
+dependencies. This can be useful in situations where one of the other processors
+doesn't provide the functionality you need to filter events.
+
+The processor can be configured by embedding Javascript in your configuration
+file or by pointing the processor at external file(s).
+
+[source,yaml]
+----
+processors:
+- script:
+    type: javascript
+    id: my_filter
+    code: >
+      function process(event) {
+          event.Tag("js");
+      }
+----
+
+This loads `filter.js` from disk.
+
+[source,yaml]
+----
+processors:
+- script:
+    type: javascript
+    id: my_filter
+    file: ${path.config}/filter.js
+----
+
+Parameters can be passed to the script by adding `script_params` to the config.
+This allows for a script to be made reusable. When using `script_params` the
+code must define a `register(params)` function to receive the parameters.
+
+[source,yaml]
+----
+processors:
+- script:
+    type: javascript
+    id: my_filter
+    script_params:
+      threshold: 15
+    code: >
+      var params = {threshold: 42};
+      function register(scriptParams) {
+          params = scriptParams;
+      }
+      function process(event) {
+          if (event.Get("severity") < params.threshold) {
+              event.Cancel();
+          }
+      }
+----
+
+If the script defines a `test()` function it will be invoked when the processor
+is loaded. Any exceptions thrown will cause the processor to fail to load. This
+can be used to make assertions about the behavior of the script.
+
+[source,javascript]
+----
+function process(event) {
+    if (event.Get("event.code") === 1102) {
+        event.Put("event.action", "cleared");
+    }
+}
+
+function test() {
+    var event = process(new Event({event: {code: 1102}));
+    if (event.Get("event.action") !== "cleared") {
+        throw "expected event.action === cleared";
+    }
+}
+----
+
+[float]
+==== Configuration options
+
+The `script` processor has the following configuration settings:
+
+`type`:: This field is required and its value must be `javascript`.
+
+`id`:: This is an optional identifier that is added to log messages. If defined
+it enables metrics logging for this instance of the processor. The metrics
+include the number of exceptions and a histogram of the execution times for
+the `process` function.
+
+`code`:: Inline Javascript code.
```

Comment:
@andrewkroh I would consider ingest node options instead of the ruby filter, I really hope that we have a painless implementation and we keep it consistent.

Reply:
I changed this to follow the ingest node script processor naming conventions.